src/main/java/com/minecolonies/structures/helpers/StructureProxy.java

```diff
@@ -365,23 +263,35 @@ public void rotate(final int times)
         minY = Math.abs(minY);
         minZ = Math.abs(minZ);
         boolean foundHut = false;
+        PlacementSettings settings = new PlacementSettings().setRotation(rotation);
 
-        for (final Template.BlockInfo info : structure.getBlockInfoWithSettings(new PlacementSettings().setRotation(rotation)))
+        for (final Template.BlockInfo info : structure.getBlockInfoWithSettings(settings))
         {
             final BlockPos tempPos = info.pos;
             final int x = tempPos.getX() + minX;
             final int y = tempPos.getY() + minY;
             final int z = tempPos.getZ() + minZ;
 
-            this.blocks[x][y][z] = info.blockState.getBlock();
-            this.metadata[x][y][z] = info.blockState;
+            this.blocks[x][y][z] = info;
+            this.entities[x][y][z] = null;
 
             if (info.blockState.getBlock() instanceof AbstractBlockHut)
             {
                 foundHut = true;
                 offset = info.pos.add(minX, minY, minZ);
             }
         }
+
+        for(final Template.EntityInfo info: structure.getTileEntities())
+        {
+            Template.EntityInfo newInfo = structure.transformEntityInfoWithSettings(info, world, Template.transformedBlockPos(settings, info.blockPos), settings);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'newInfo' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)